### PR TITLE
Woo: Add product variation API state/handling

### PR DESCRIPTION
**Pull request by coderkevin**
This adds code to support creating product variations, and updating them in the local state.

To Test: `npm test`

### CODE REVIEW

**Comment by coderkevin** [comment]
Rebased.

**Comment by coderkevin** [comment]
Hold off on reviewing this one as I need to adjust where the reducer stores state. I'll update afterwards.

**Comment by coderkevin** [comment]
Okay, this is good to review again. I moved the product variations state code to directly under `state/sites/product-variations` now.

File: client/extensions/woocommerce/state/data-layer/product-variations/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * External dependencies
+ */
+import { isNumber } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchWithProps } from 'woocommerce/state/helpers';
+import { post } from 'woocommerce/state/data-layer/request/actions';
+import { setError } from 'woocommerce/state/sites/status/wc-api/actions';
```
Comment (justinshreve) [inline client/extensions/woocommerce/state/data-layer/product-variations/index.js]: Minor, import out of alphabetical order.

**Pull request merged**